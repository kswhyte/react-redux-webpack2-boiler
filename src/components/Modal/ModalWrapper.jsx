import React from 'react';
import propTypes from 'prop-types';

import './modal.css';

const ModalWrapper = props => {

  const handleBackgroundClick = e => {
    if (e.target === e.currentTarget) props.hideModal();
  };



  return (
  <section>
    <div onClick={handleBackgroundClick} className={`modal fade in ${props.modalType}`} tabIndex="-1" role="dialog">
      <div className={"modal-dialog"} role="document">
        <div className={"modal-content"}>
          <div className={"modal-header"}>
            <button onClick={props.hideModal} type="button" className={"close"} aria-label="Close"><span aria-hidden="true">&times;</span></button>
            <h4 className={"modal-title"}>{props.title}</h4>
          </div>
          <div className={"modal-body"}>
            {props.children}
          </div>
        </div>
      </div>
    </div>
    <div className="modal-background"></div>
  </section>
  );
};

ModalWrapper.propTypes = {
  // props
  title: propTypes.string,
  showOk: propTypes.bool,
  okText: propTypes.string,
  modalType: propTypes.string,
  okDisabled: propTypes.bool,
  width: propTypes.number,
  style: propTypes.object,
  children: propTypes.oneOfType([
    propTypes.array,
    propTypes.element,
    propTypes.string,
  ]).isRequired,

  // methods
  hideModal: propTypes.func
};

ModalWrapper.defaultProps = {
  title: '',
  showOk: true,
  okText: 'OK',
  okDisabled: false,
  width: 400,
  hideModal: () => {}
};

export default ModalWrapper;
