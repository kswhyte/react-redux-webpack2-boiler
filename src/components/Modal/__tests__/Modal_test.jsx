/*
This file does not need expect.
jest has expect built in.
These test are not for storybook : we can talk about that

*/

/*
Note from Danny - I set this up for storybook because it was breaking.
We could alternatively rename this file so that storybook vs jest test are different
*/

import React from 'react';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import renderer from 'react-test-renderer';
import ModalWrapper from '../ModalWrapper';
import LogOutAreYouSure from '../components/logoutAreYouSure';

import {mount} from 'enzyme';


const stories = storiesOf('ModalWrapper', module);

stories.add('Modal Story', () => {
    const modalStory = <ModalWrapper>children</ModalWrapper>;
    specs(() =>
    describe('Show NotFound', () => {
        it('Should render the NotFound component without crashing', () => {
            let output = mount(modalStory);
            expect(output.find('.modal-dialog').length).toEqual(1);
          });
        it('renders Wrapper Correctly', () => {
          const tree = renderer.create(
            <ModalWrapper>children</ModalWrapper>
          ).toJSON();
          expect(tree).toMatchSnapshot();
        });

        it('renders Logout Modal correctly', () => {
          const tree = renderer.create(
            <LogOutAreYouSure/>
          ).toJSON();
          expect(tree).toMatchSnapshot();
        });
    }))
    return modalStory;
});





