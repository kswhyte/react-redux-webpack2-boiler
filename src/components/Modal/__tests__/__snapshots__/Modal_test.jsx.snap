// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Story 1`] = `
<section>
  <div
    className="modal fade in undefined"
    onClick={[Function]}
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
          >
            
          </h4>
        </div>
        <div
          className="modal-body"
        >
          children
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-background"
  />
</section>
`;

exports[`Show NotFound renders Logout Modal correctly 1`] = `
<section>
  <div
    className="modal fade in undefined"
    onClick={[Function]}
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
          >
            Are you sure you want to log out?
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <p>
            You are in the middle of a session and all unsaved changes may be lost
          </p>
          <div
            className="row"
          >
            <div
              className="col-xs-12"
            >
              <div
                className="pull-right"
              >
                <button
                  className="btn-default btn"
                  onClick={undefined}
                >
                  Cancel
                </button>
                <button
                  className="btn-primary btn"
                  onClick={undefined}
                >
                  LogOut
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-background"
  />
</section>
`;

exports[`Show NotFound renders Wrapper Correctly 1`] = `
<section>
  <div
    className="modal fade in undefined"
    onClick={[Function]}
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
          >
            
          </h4>
        </div>
        <div
          className="modal-body"
        >
          children
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-background"
  />
</section>
`;
